extension/src-language-server/fta/diagram/fta-diagram-generator.ts

```diff
@@ -70,14 +104,60 @@ export class FtaDiagramGenerator extends LangiumDiagramGenerator {
     protected generateEdges(node: AstNode, idCache: IdCache<AstNode>): SModelElement[] {
         const elements: SModelElement[] = [];
         const sourceId = idCache.getId(node);
-        // for every reference an edge is created
-        const targets = getTargets(node);
-        for (const target of targets) {
-            const targetId = idCache.getId(target);
-            const edgeId = idCache.uniqueId(`${sourceId}_${targetId}`, undefined);
-            if (sourceId && targetId) {
-                const e = this.generateFTEdge(edgeId, sourceId, targetId, idCache);
-                elements.push(e);
+        if (sourceId) {
+            // for every reference an edge is created
+            const targets = getTargets(node);
+            for (const target of targets) {
+                const targetId = idCache.getId(target);
+                const edgeId = idCache.uniqueId(`${sourceId}_${targetId}`, undefined);
+
+                // create port for the source node
+                const sourceNode = this.idToSNode.get(sourceId);
+                const sourcePortId = idCache.uniqueId(edgeId + "_port");
+                sourceNode?.children?.push(this.createFTAPort(sourcePortId, PortSide.SOUTH));
+
+                // create port for source parent and edge to this port
+                let sourceParentPortId: string | undefined = undefined;
```

Comment:
wouldn't it be undefined anyway if you just declare it and don't assign it undefined explicitly